master/internal/command/authz_basic_impl.go

```diff
@@ -11,11 +11,29 @@ type NSCAuthZBasic struct{}
 
 // CanGetNSC returns a nil error.
 func (a *NSCAuthZBasic) CanGetNSC(
-	ctx context.Context, curUser model.User, ownerID model.UserID, workspaceID model.AccessScopeID,
+	ctx context.Context, curUser model.User, workspaceID model.AccessScopeID,
 ) (canGetCmd bool, serverError error) {
 	return true, nil
 }
 
+// FilterNSCWorkspaces determines which workspaces out of the given workspaces
+// the user has view access to.
+func (a *NSCAuthZBasic) FilterNSCWorkspaces(
+	ctx context.Context, curUser model.User, workspaceSet model.AccessScopeSet,
+) (model.AccessScopeSet, error) {
+	return workspaceSet, nil
+}
+
+// AccessibleScopes returns the set of scopes that the user should be limited to.
+func (a *NSCAuthZBasic) AccessibleScopes(
+	ctx context.Context, curUser model.User, requestedScope model.AccessScopeID,
+) (*model.AccessScopeSet, error) {
+	if requestedScope == 0 {
+		return nil, nil
+	}
+	return &model.AccessScopeSet{requestedScope: true}, nil
```

Comment:
wait sorry i'm revoking my approval for now.  I think we should decide right now if an empty access scope set should mean none or all accessible. If it means none are accessible, we need to return a set of all workspace ids.

Reply:
I think the best way to do this is just to do a db query on workspaces and return all of them

Reply:
I think what I was going with was:
- nil: no access control applies
- the zero value: no accessible scopes
- other cases: exact workspace(s) you have access to or asked for 
 
this avoid that db query in OSS but it's less intuitive

Reply:
ok in that case, I think we should return a `*model.AccessScopeID` value instead of a set. That way it can convey all the same information and we don't have to define a new type

Reply:
I think that would the case of 

> other cases: exact workspace(s) you have access to or asked for

the plural case for this where the user doesn't ask for a particular workspace (or asks for a list of workspace which we don't support atm) but they have access to 2+ workspaces.